TodoApp.ConsoleApp/Services/TodoService.cs

```diff
@@ -1,77 +1,92 @@
 
-            IEnumerable<Task> tasks = reader.ReadTasks();
+        public TodoService(string path)
+        {
+            Path = path;
 
-            return tasks;
+            if (!File.Exists(Path))
+            {
+                XmlWriterSettings xmlWriterSettings = new XmlWriterSettings();
+                xmlWriterSettings.Indent = true;
+                Encoding encoding = Encoding.GetEncoding("UTF-8");
+                using (XmlWriter writer = XmlWriter.Create(Path))
+                {
+                    writer.WriteStartDocument();
+                    writer.WriteStartElement("todos");
+                    writer.WriteEndElement();
+                    writer.WriteEndDocument();
+                    writer.Close();
+                }
+            }
         }
 
-        public Task GetByID(Guid id)
+        public IEnumerable<TodoModel> GetAll()
         {
-            IEnumerable<Task> tasks = GetAll();
-
-            Task wantedTask = null;
-
-            wantedTask = tasks.FirstOrDefault(task => task.ID == id);
-
-            return wantedTask;
+            return repo.GetAll();
         }
 
-        public bool Update(Task task)
+        public TodoModel GetByID(Guid id)
         {
-            if (task == null)
-            {
-                throw new ArgumentNullException(nameof(task));
-            }
-
-            bool isDeleted = Delete(task.ID);
+            return repo.GetById(id);
+        }
 
-            if (!isDeleted)
+        public bool Update(TodoModel model)
+        {
+            if (model == null)
             {
-                return false;
+                throw new ArgumentNullException(nameof(model));
             }
 
-            var newTask = Create(task);
+            repo.Update(model);
+            repo.Save();
 
-            return newTask != null;
+            return model != null;
         }
 
-        public Task Create(Task task)
+        public TodoModel Create(TodoModel model)
         {
-            XMLTaskWriter.Save(task);
+            repo.Insert(model);
+            repo.Save();
 
-            return task;
+            return model;
         }
 
         public bool Delete(Guid id)
         {
-            Task taskToBeDeleted = GetByID(id);
+            TodoModel modelToBeDeleted = GetByID(id);
             
-            if (taskToBeDeleted == null)
+            if (modelToBeDeleted == null)
             {
                 return false;
             }
 
-            XMLTaskWriter.Delete(taskToBeDeleted);
+            repo.Delete(id);
+            repo.Save();
+
             return true;
         }
 
         public bool DeleteByIndex(int index)
         {
-            var tasks = this.GetAll();
-            var taskId = tasks.ElementAt(index).ID;
-            return this.Delete(taskId);
+            var models = repo.GetAll();
+            var modelId = models.ElementAt(index).Id;
+            bool isDeleted = this.Delete(modelId);
+            repo.Save();
```

Comment:
`this.Delete(modelId)` already saves the data, so in this case it is not required to call `repo.Save()` again.